pkg/apis/core/validation/validation_test.go

```diff
@@ -10552,29 +10552,417 @@ func TestValidateServiceCreate(t *testing.T) {
 			},
 			numErrs: 1,
 		},
+		/* ip families validation */
+		{
+			name: "invalid, service with invalid ipFamilies",
+			tweakSvc: func(s *core.Service) {
+				invalidServiceIPFamily := core.IPFamily("not-a-valid-ip-family")
+				s.Spec.IPFamilies = []core.IPFamily{invalidServiceIPFamily}
+			},
+			numErrs: 1,
+		},
+		{
+			name: "invalid, service with invalid ipFamilies (2nd)",
+			tweakSvc: func(s *core.Service) {
+				invalidServiceIPFamily := core.IPFamily("not-a-valid-ip-family")
+				s.Spec.IPFamilyPolicy = &requireDualStack
+				s.Spec.IPFamilies = []core.IPFamily{core.IPv4Protocol, invalidServiceIPFamily}
+			},
+			numErrs:              1,
+			dualStackFeatureGate: true,
+		},
+		{
+			name: "invalid, wrong IPFamilyPolicy(singleStack) is set for two families",
+			tweakSvc: func(s *core.Service) {
+				s.Spec.IPFamilyPolicy = &singleStack
+				s.Spec.IPFamilies = []core.IPFamily{core.IPv4Protocol, core.IPv6Protocol}
+			},
+			numErrs:              1,
+			dualStackFeatureGate: true,
+		},
+		{
+			name: "valid, IPFamilyPolicy(preferDualStack) is set for two families (note: alloc sets families)",
```

Comment:
shouldn't this fail validation - doesn't it mean RequireDualStack?

Edit: oh, updates after being allocated.   Hmm.  If I say "Prefer" and then spec both families, and my cluster is only single-stack - is it an error?

Reply:
Prefer + two families (as provided by user) will fail validation (allocation code does validate for that).